client/src/pages/student/Registration.js

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import * as yup from 'yup';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+export default function Registration() {
+
+  function registration(data) {
+
+  }
+
+  const schema = yup.object().shape({
+    name: yup.string().matches(/^[A-Za-z ]*$/, 'Please enter valid name').required("Name field cannot be empty*"),
+    email: yup.string().required("Email is required*").matches(/^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/g, 'Email must be valid'),
+    mobile: yup.string().required("Mobile is required*").min(10,'Mobile number must be 10 digits').max(10,'Mobile number must be 10 digits').matches(/^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/,'Enter valid mobile number'),
+    password: yup.string().required("Password cannot be empty*").matches(/^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,'Minimum eight characters, at least one letter and one number'),
```

Comment:
Use single quotes to represent string values.

Reply:
Added single quotes to represent string value.